File: Scripts/results/generated_code_results/Flutter/Education/OnlineTeachingSoftware/DiscoverScreen.dart
import 'package:flutter/material.dart';



class DiscoverScreen extends StatelessWidget {

  @override

  Widget build(BuildContext context) {

    return Scaffold(

      body: SafeArea(

        child: Column(

          children: [

            _buildHeader(context),

            Expanded(

              child: SingleChildScrollView(

                child: Column(

                  crossAxisAlignment: CrossAxisAlignment.start,

                  children: [

                    _buildHeroSection(),

                    _buildFeaturedCourse(context),

                    _buildPopularCourses(context),

                  ],

                ),

              ),

            ),

            _buildBottomNavBar(context),

          ],

        ),

      ),

    );

  }



  Widget _buildHeader(BuildContext context) {

    return Padding(

      padding: const EdgeInsets.all(16.0),

      child: Row(

        mainAxisAlignment: MainAxisAlignment.spaceBetween,

        children: [

          IconButton(

            icon: Icon(Icons.grid_view),

            onPressed: () {},

          ),

          Text(

            'Discover',

            style: TextStyle(fontSize: 20, fontWeight: FontWeight.bold),

          ),

          IconButton(

            icon: Stack(

              children: [

                Icon(Icons.notifications),

                Positioned(

                  right: 0,

                  top: 0,

                  child: Container(

                    padding: EdgeInsets.all(2),

                    decoration: BoxDecoration(

                      color: Colors.red,

                      borderRadius: BorderRadius.circular(6),

                    ),

                    constraints: BoxConstraints(

                      minWidth: 12,

                      minHeight: 12,

                    ),

                  ),

                ),

              ],

            ),

            onPressed: () => Navigator.pushNamed(context, 'Notifications'),

          ),

        ],

      ),

    );

  }



  Widget _buildHeroSection() {

    return Padding(

      padding: const EdgeInsets.all(16.0),

      child: Text(

        'Study what you want anytime anywhere',

        style: TextStyle(fontSize: 24, fontWeight: FontWeight.bold),

      ),

    );

  }



  Widget _buildFeaturedCourse(BuildContext context) {

    return Padding(

      padding: const EdgeInsets.all(16.0),

      child: GestureDetector(

        onTap: () => Navigator.pushNamed(context, 'CourseDetail'),

        child: Container(

          decoration: BoxDecoration(

            color: Colors.white,

            borderRadius: BorderRadius.circular(12),

            boxShadow: [

              BoxShadow(

                color: Colors.grey.withOpacity(0.1),

                spreadRadius: 1,

                blurRadius: 5,

                offset: Offset(0, 3),

              ),

            ],

          ),

          child: Column(

            crossAxisAlignment: CrossAxisAlignment.start,

            children: [

              ClipRRect(

                borderRadius: BorderRadius.vertical(top: Radius.circular(12)),

                child: Image.asset('assets/images/template.png', fit: BoxFit.cover),

              ),

              Padding(

                padding: const EdgeInsets.all(12.0),

                child: Column(

                  crossAxisAlignment: CrossAxisAlignment.start,

                  children: [

                    Text('Interface Design', style: TextStyle(color: Colors.blue)),

                    SizedBox(height: 4),

                    Text(

                      'User interface for beginners',

                      style: TextStyle(fontSize: 18, fontWeight: FontWeight.bold),

                    ),

                    SizedBox(height: 8),

                    Row(

                      children: [

                        Icon(Icons.star, color: Colors.yellow, size: 16),

                        Text(' 4.7 (1,242 Reviews)'),

                      ],

                    ),

                    SizedBox(height: 8),

                    LinearProgressIndicator(

                      value: 0.48,

                      backgroundColor: Colors.grey[200],

                      valueColor: AlwaysStoppedAnimation<Color>(Colors.blue),

                    ),

                    SizedBox(height: 4),

                    Text('48% completed', style: TextStyle(color: Colors.grey)),

                  ],

                ),

              ),

            ],

          ),

        ),

      ),

    );

  }



  Widget _buildPopularCourses(BuildContext context) {

    return Column(

      crossAxisAlignment: CrossAxisAlignment.start,

      children: [

        Padding(

          padding: const EdgeInsets.symmetric(horizontal: 16.0),

          child: Row(

            mainAxisAlignment: MainAxisAlignment.spaceBetween,

            children: [

              Text(

                'Popular Course',

                style: TextStyle(fontSize: 20, fontWeight: FontWeight.bold),

              ),

              TextButton(

                onPressed: () {},

                child: Text('See All'),

              ),

            ],

          ),

        ),

        SizedBox(height: 8),

        SingleChildScrollView(

          scrollDirection: Axis.horizontal,

          child: Row(

            children: [

              _buildCourseCard(

                context,

                'Graphic Design',

                'Expert Wireframing for Mobile Design',

                'Jerremy Mamika',

                '\$48',

              ),

              _buildCourseCard(

                context,

                'Development',

                'Bootstrapping Principles',

                'Alexander Be',

                '\$55',

              ),

            ],

          ),

        ),

      ],

    );

  }



  Widget _buildCourseCard(BuildContext context, String category, String title, String instructor, String price) {

    return GestureDetector(

      onTap: () => Navigator.pushNamed(context, 'CourseDetail'),

      child: Container(

        width: 240,

        margin: EdgeInsets.only(left: 16, bottom: 16),

        decoration: BoxDecoration(

          color: Colors.white,

          borderRadius: BorderRadius.circular(12),

          boxShadow: [

            BoxShadow(

              color: Colors.grey.withOpacity(0.1),

              spreadRadius: 1,

              blurRadius: 5,

              offset: Offset(0, 3),

            ),

          ],

        ),

        child: Column(

          crossAxisAlignment: CrossAxisAlignment.start,

          children: [

            Stack(

              children: [

                ClipRRect(

                  borderRadius: BorderRadius.vertical(top: Radius.circular(12)),

                  child: Image.asset('assets/images/template.png', height: 120, width: 240, fit: BoxFit.cover),

                ),

                Positioned(

                  top: 8,

                  right: 8,

                  child: IconButton(

                    icon: Icon(Icons.favorite_border, color: Colors.white),

                    onPressed: () {},

                  ),

                ),

              ],

            ),

            Padding(

              padding: const EdgeInsets.all(12.0),

              child: Column(

                crossAxisAlignment: CrossAxisAlignment.start,

                children: [

                  Text(category, style: TextStyle(color: Colors.blue)),

                  SizedBox(height: 4),

                  Text(

                    title,

                    style: TextStyle(fontSize: 16, fontWeight: FontWeight.bold),

                    maxLines: 2,

                    overflow: TextOverflow.ellipsis,

                  ),

                  SizedBox(height: 4),

                  Text(instructor, style: TextStyle(color: Colors.grey)),

                  SizedBox(height: 4),

                  Text(price, style: TextStyle(fontWeight: FontWeight.bold)),

                ],

              ),

            ),

          ],

        ),

      ),

    );

  }



  Widget _buildBottomNavBar(BuildContext context) {

    return BottomNavigationBar(

      currentIndex: 0,

      onTap: (index) {

        if (index == 1) {

          Navigator.pushNamed(context, 'Statistics');

        } else if (index == 2) {

          Navigator.pushNamed(context, 'Favorites');

        }

      },

      items: [

        BottomNavigationBarItem(icon: Icon(Icons.home), label: 'Home'),

        BottomNavigationBarItem(icon: Icon(Icons.bar_chart), label: 'Statistic'),

        BottomNavigationBarItem(icon: Icon(Icons.favorite), label: 'Favorites'),

        BottomNavigationBarItem(icon: Icon(Icons.person), label: 'Profile'),

      ],

    );

  }

}

File: Scripts/results/generated_code_results/RN_gpt/Education/OnlineTeachingSoftware/DiscoverScreen.js
import React from 'react';
import { View, Text, StyleSheet, Image, Button, TouchableOpacity, StatusBar } from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';

const DiscoverScreen = ({ navigation }) => {
  return (
    <View style={styles.container}>
      <StatusBar barStyle="light-content" />
      <View style={styles.header}>
        <TouchableOpacity>
          <Icon name="grid-outline" size={24} color="#fff" />
        </TouchableOpacity>
        <Text style={styles.headerTitle}>Discover</Text>
        <TouchableOpacity onPress={() => navigation.navigate('Notifications')}>
          <Icon name="notifications-outline" size={24} color="#fff" />
        </TouchableOpacity>
      </View>

      <Text style={styles.heroTitle}>Study what you want anytime anywhere</Text>

      <TouchableOpacity style={styles.featuredCourse} onPress={() => navigation.navigate('CourseDetail')}>
        <Text style={styles.courseCategory}>Interface Design</Text>
        <Text style={styles.courseTitle}>User interface for beginners</Text>
        <View style={styles.ratingContainer}>
          <Text style={styles.ratingText}>⭐ 4.7 (1242 Reviews)</Text>
        </View>
        <View style={styles.progress}>
          <View style={styles.progressBar}></View>
          <Text style={styles.progressText}>48%</Text>
        </View>
        <Image source={require('../assets/snack-icon.png')} style={styles.courseImage} />
      </TouchableOpacity>

      <View style={styles.sectionHeader}>
        <Text style={styles.sectionTitle}>Popular Course</Text>
        <TouchableOpacity>
          <Text style={styles.seeAll}>See All</Text>
        </TouchableOpacity>
      </View>

      <View style={styles.courseList}>
        <TouchableOpacity style={styles.courseCard} onPress={() => navigation.navigate('CourseDetail')}>
          <Image source={require('../assets/snack-icon.png')} style={styles.courseCardImage} />
          <View style={styles.courseInfo}>
            <Text style={styles.courseCardCategory}>Graphic Design</Text>
            <Text style={styles.courseCardTitle}>Expert Wireframing for Mobile Design</Text>
            <Text style={styles.courseCardInstructor}>Jerremy Mamika</Text>
            <Text style={styles.courseCardPrice}>$48</Text>
          </View>
          <TouchableOpacity onPress={() => navigation.navigate('Favorites')}>
            <Icon name="heart-outline" size={24} color="#000" style={styles.favoriteIcon} />
          </TouchableOpacity>
        </TouchableOpacity>
        <TouchableOpacity style={styles.courseCard} onPress={() => navigation.navigate('CourseDetail')}>
          <Image source={require('../assets/snack-icon.png')} style={styles.courseCardImage} />
          <View style={styles.courseInfo}>
            <Text style={styles.courseCardCategory}>Development</Text>
            <Text style={styles.courseCardTitle}>Bootstrapping Principles</Text>
            <Text style={styles.courseCardInstructor}>Alexander Be</Text>
          </View>
        </TouchableOpacity>
      </View>

      <View style={styles.bottomNav}>
        <TouchableOpacity>
          <View style={styles.navItem}>
            <Icon name="home-outline" size={24} color="#6200ee" />
            <Text style={styles.navText}>Home</Text>
          </View>
        </TouchableOpacity>
        <TouchableOpacity onPress={() => navigation.navigate('Statistics')}>
          <View style={styles.navItem}>
            <Icon name="stats-chart-outline" size={24} color="#aaa" />
            <Text style={styles.navText}>Statistic</Text>
          </View>
        </TouchableOpacity>
        <TouchableOpacity onPress={() => navigation.navigate('Favorites')}>
          <View style={styles.navItem}>
            <Icon name="heart-outline" size={24} color="#aaa" />
            <Text style={styles.navText}>Favorites</Text>
          </View>
        </TouchableOpacity>
        <TouchableOpacity>
          <View style={styles.navItem}>
            <Icon name="person-outline" size={24} color="#aaa" />
            <Text style={styles.navText}>Profile</Text>
          </View>
        </TouchableOpacity>
      </View>
    </View>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#f5f5f5',
  },
  header: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    backgroundColor: '#6200ee',
    paddingHorizontal: 20,
    paddingTop: 50,
    paddingBottom: 20,
  },
  headerTitle: {
    color: '#fff',
    fontSize: 18,
  },
  heroTitle: {
    textAlign: 'center',
    fontSize: 16,
    marginVertical: 20,
    color: '#333',
  },
  featuredCourse: {
    backgroundColor: '#fff',
    marginHorizontal: 20,
    borderRadius: 10,
    padding: 15,
    elevation: 2,
  },
  courseCategory: {
    fontSize: 12,
    color: '#999',
  },
  courseTitle: {
    fontSize: 16,
    color: '#333',
    marginVertical: 5,
  },
  ratingContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    marginVertical: 5,
  },
  ratingText: {
    fontSize: 12,
    color: '#999',
  },
  progress: {
    flexDirection: 'row',
    alignItems: 'center',
    marginVertical: 5,
  },
  progressBar: {
    flex: 1,
    height: 5,
    backgroundColor: '#6200ee',
    borderRadius: 2.5,
    marginRight: 10,
  },
  progressText: {
    fontSize: 12,
    color: '#333',
  },
  courseImage: {
    width: 50,
    height: 50,
    position: 'absolute',
    top: 15,
    right: 15,
  },
  sectionHeader: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    marginHorizontal: 20,
    marginVertical: 15,
  },
  sectionTitle: {
    fontSize: 16,
    color: '#333',
  },
  seeAll: {
    fontSize: 12,
    color: '#6200ee',
  },
  courseList: {
    marginHorizontal: 20,
  },
  courseCard: {
    backgroundColor: '#fff',
    flexDirection: 'row',
    borderRadius: 10,
    marginBottom: 15,
    overflow: 'hidden',
    elevation: 2,
  },
  courseCardImage: {
    width: 100,
    height: 100,
  },
  courseInfo: {
    flex: 1,
    padding: 10,
  },
  courseCardCategory: {
    fontSize: 12,
    color: '#6200ee',
  },
  courseCardTitle: {
    fontSize: 14,
    color: '#333',
    marginVertical: 5,
  },
  courseCardInstructor: {
    fontSize: 12,
    color: '#999',
  },
  courseCardPrice: {
    fontSize: 12,
    color: '#333',
    marginVertical: 5,
  },
  favoriteIcon: {
    position: 'absolute',
    top: 10,
    right: 10,
  },
  bottomNav: {
    position: 'absolute',
    bottom: 0,
    left: 0,
    right: 0,
    height: 80,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    backgroundColor: '#fff',
    paddingHorizontal: 40,
    elevation: 10,
  },
  navItem: {
    alignItems: 'center',
  },
  navText: {
    fontSize: 10,
    color: '#777',
  },
});

export default DiscoverScreen;